app/javascript/gobierto_dashboards/webapp/containers/subsidies/Summary.vue

```diff
@@ -158,13 +188,13 @@ export default {
       labelMainBeneficiaries: I18n.t('gobierto_dashboards.dashboards.subsidies.main_beneficiaries')
     }
   },
-
   mounted() {
     EventBus.$on('refresh-summary-data', () => {
       this.refreshSummaryData();
     });
 
     EventBus.$emit("summary-ready");
+    EventBus.$on('filtered-items-grouped', (data, value) => this.updateItems(data, value))
```

Comment:
Actually subsidies and contracts are really really really similar. As a first approach, you should move the shared things to a `mixin`.
But, better than that, it seems the major difference is about what property (_beneficiary_name_ vs _assignee_) is dealing with. Why don't create a component to handle it no matter what the data contains?